### PR TITLE
Convert `TokenField.unit.spec.js` from Enzyme to RTL

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
PENDING REVIEW

### What does this PR accomplish?
- Converts `TokenField.unit.spec.js` from using Enzyme to RTL
- Merges into #14094 - the whole point of this commit is to make it easier for review, since this file is huge
- Adds 2 new libraries (more info [here](https://github.com/testing-library/user-event#installation))
```
yarn add @testing-library/user-event @testing-library/dom --dev
```

### Additional notes:
- There are some tests that are still skipped
    - Either I couldn't confirm that they are working as they should or (more likely) I had trouble using the tools from RTL to achieve the desired result
- Since this is branched of the `remove-enzyme-unit` branch, one unit test will fail in CI (it's still not fixed there), but it shouldn't be related to `TokendField.unit.spec.js`

### CODE REVIEW

File: frontend/test/metabase/components/TokenField.unit.spec.js [hunk 1]
```diff
@@ -195,118 +188,122 @@ describe("TokenField", () => {
...
-      keyDown(KEYCODE_ENTER);
-      expect(values()).toEqual(["Gadget"]);
-      expect(options()).toEqual(["Doohickey", "Gizmo", "Widget"]);
+      fireEvent.change(input(), { target: { value: "Gad" } });
+      within(options()).getByText("Gadget");
+      fireEvent.keyDown(input(), { keyCode: KEYCODE_ENTER });
+      within(values()).getByText("Gadget");
+      within(options()).getByText("Doohickey");
+      within(options()).getByText("Gizmo");
+      within(options()).getByText("Widget");
```
Comment (paulrosenzweig) [inline frontend/test/metabase/components/TokenField.unit.spec.js:281]: Here, it's kinda nice that the old assertions are more specific. 

`expect(options()).toEqual(["Doohickey", "Gizmo", "Widget"]);` also asserts that "Gadget" isn't in the list. Would it be possible to add a helper that also checks how many items are in the list? 
e.g. `optionsContains(["Doohickey", "Gizmo", "Widget"])`

Maybe looking at something like `within(options()).getAllByText(/./).length`?

**Review by paulrosenzweig — APPROVED** [review]
Looks great!

I think maybe a few more helpers could be nice, but overall I think it's awesome.

I think commenting out the tricky to test cases makes sense for now.

File: frontend/test/metabase/components/TokenField.unit.spec.js [hunk 2]
```diff
@@ -195,118 +188,122 @@ describe("TokenField", () => {
...
 
+    // This is messy and tricky to test with RTL
     it("should not commit empty freeform value", () => {
-      focusAndType("Doohickey");
-      focusAndType("");
-      blur();
-      expect(value()).toEqual([]);
-      expect(values()).toEqual([]);
+      fireEvent.change(input(), { target: { value: "Doohickey" } });
+      userEvent.clear(input());
```
[the remark this answers is not in the document]
Comment (nemanjaglumac) [reply frontend/test/metabase/components/TokenField.unit.spec.js:226]: Unfortunately, no.
Ideally, it should be possible to accomplish all this with `fireEvent` only but I couldn't find a way to do it.

File: frontend/test/metabase/components/TokenField.unit.spec.js [hunk 3]
```diff
@@ -62,128 +61,122 @@ class TokenFieldWithStateAndDefaults extends React.Component {
...
 describe("TokenField", () => {
-  let component;
-  const input = () => component.find("input");
-  const value = () => component.state().value;
-  const options = () => component.find(MockOption).map(o => o.text());
-  const values = () => component.find(MockValue).map(v => v.text());
-  const blur = () => input().simulate("blur");
-  const focus = () => input().simulate("focus");
-  const type = str => input().simulate("change", { target: { value: str } });
-  const focusAndType = str => focus() && type(str);
```
[the remark this answers is not in the document]
Comment (nemanjaglumac) [reply frontend/test/metabase/components/TokenField.unit.spec.js:73]: Valid point. Tbh, I was thinking about it but got overwhelmed by the sheer number of tests in this file so I focused my attention on each test individually. By the time I finished I forgot to tidy-up everything.

File: frontend/test/metabase/components/TokenField.unit.spec.js (hunk 1 above)
Comment (nemanjaglumac) [reply frontend/test/metabase/components/TokenField.unit.spec.js:281]: I'll see what can I do about it. There must be some way to check this. What I know for sure is that we can check for `textContent` but it concatenates all text content. In other words, if you want to check if options contain `["Doohickey", "Gizmo", "Widget"]`, the text content would give you `DoohickeyGizmoWidget`. If you're ok with it, I can use that approach?

Comment (paulrosenzweig) [reply frontend/test/metabase/components/TokenField.unit.spec.js:281]: 👍 Whatever works! The helper will only be used in this file so I don't think it needs to be elegant as long as it asserts the right thing.

File: frontend/test/metabase/components/TokenField.unit.spec.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (nemanjaglumac) [reply frontend/test/metabase/components/TokenField.unit.spec.js:73]: Added helpers in 2a05e29158b096514d353ab1a4b161f7aff0b403. Not only did it help to write more precise and succinct code, but it also removed the need to check for non-existence of some elements.

File: frontend/test/metabase/components/TokenField.unit.spec.js (hunk 1 above)
Comment (nemanjaglumac) [reply frontend/test/metabase/components/TokenField.unit.spec.js]: Added helpers in 2a05e29158b096514d353ab1a4b161f7aff0b403. Not only did it help to write more precise and succinct code, but it also removed the need to check for non-existence of some elements.

**Review by paulrosenzweig — APPROVED** [review]
New helpers look great!

**Pull request merged**